pkg/http/request/request.go

```diff
@@ -12,51 +14,363 @@ import (
 
 // ExtractModelFromRequest extracts the model from the request body.
 func ExtractModelFromRequest(req *http.Request) string {
+	if req == nil {
+		slog.Error("❌ Request is nil in ExtractModelFromRequest")
+		return ""
+	}
+
+	if req.Body == nil {
+		slog.Error("❌ Request body is nil in ExtractModelFromRequest")
+		return ""
+	}
+
 	body, err := io.ReadAll(req.Body)
 	if err != nil {
+		slog.Error("❌ Failed to read body in ExtractModelFromRequest", "error", err)
 		return ""
 	}
 
+	// Always restore the body for future reads
 	req.Body = io.NopCloser(bytes.NewBuffer(body))
 
+	// Handle empty body
+	if len(body) == 0 {
+		slog.Error("❌ Empty request body in ExtractModelFromRequest")
+		return ""
+	}
```

Comment:
For these various empty strings, please remind me: are you raising errors downstream, or also logging and suppressing?

Reply:
yup, this was checked on a higher level, but to make it safer, I added a return of (string, error) and check everytime I use this function. Also I updated the tests for this. 

Thanks 🙏